### PR TITLE
Add support for posting metadata

### PR DESCRIPTION
**Pull request by cruizh**
I wanted to be able to use posting metadata when importing. I noticed `bean-extract` generated the poting metadata correctly, but Fava's importing feature discarded posting metadata and only kept entry metadata.

This PR adds an optional `meta` field to `Posting` so that this information is not lost.
I also modified the `Posting` entry form to include a button to add metadata to the posting and show the posting metadata. This allows the user to enter posting metadata on import and on new transactions.

It looks like this on the UI:
image

### CODE REVIEW

File: src/fava/beans/abc.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def price(self) -> Amount | None:
 
     @property
     @abstractmethod
-    def meta(self) -> Meta | None:
+    def meta(self) -> Meta:
```
Comment (yagebu) [inline src/fava/beans/abc.py:115]: this change should be reverted - we still need to support postings with a `None` meta.

File: frontend/src/entry-forms/AddPostingMetadataButton.svelte [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<script lang="ts">
```
Comment (yagebu) [inline frontend/src/entry-forms/AddPostingMetadataButton.svelte:1]: This seems to be nearly an identical copy of the corresponding component for entries - please reuse that one

File: frontend/src/entry-forms/PostingMetadata.svelte [hunk 3]
```diff
@@ -0,0 +1,107 @@
+<script lang="ts">
```
Comment (yagebu) [inline frontend/src/entry-forms/PostingMetadata.svelte:1]: This seems to be nearly an identical copy of the corresponding component for entries - please reuse that one

File: frontend/src/entries.ts [hunk 4]
```diff
@@ -15,11 +16,19 @@ import {
 export interface Posting {
   account: string;
   amount: string;
+  meta: EntryMetadata | null;
```
Comment (yagebu) [inline frontend/src/entries.ts:19]: I think dropping the `| null` here and ensuring that this is always set in the frontend code even if not set in the Beancount entries might simplify reuse of the existing components

**Review by yagebu — CHANGES_REQUESTED** [review]
Thanks for this PR :)

This looks mostly good but the duplicated code should be reduced, I've left some inline comments.

File: src/fava/beans/abc.py (hunk 1 above)
Comment (cruizh) [reply src/fava/beans/abc.py]: This should be fixed in https://github.com/beancount/fava/pull/1722/commits/33d1c8b094cd5115ff90fa3a29dd71ad70f80faa :)

File: frontend/src/entries.ts (hunk 4 above)
Comment (cruizh) [reply frontend/src/entries.ts]: This should be fixed in https://github.com/beancount/fava/pull/1722/commits/b851eb3ee76617e2d00b63980489f354356777a2 :)

**Comment by cruizh** [comment]
Thanks a lot for the quick response and the constructive criticism! I have been using Fava for a long time and appreciate all the hard work you put into it.

I have fixed the points you commented. I hope this makes it an easier change to accept.

Please let me know if I should improve it further. Again, thanks a lot!

**Review by yagebu — APPROVED** [review]

**Comment by yagebu** [comment]
Thanks again, nice work :)

**Pull request merged**